native-heapdump-example/README.md

```diff
@@ -0,0 +1,130 @@
+This provides you with enough time to send it a `SIGUSR1` signal. The application will handle the signal and create a heap dump in the application's working directory. The heap dump will contain the `Collection` of `Person`s referenced by the static variable `CROWD`.
+
+Follow these steps to build a native executable that will produce a heap dump when it receives a `SIGUSR1` signal.
+
+### Build the Application
+
+1. Compile _SVMHeapDump.java_ using `javac`:
+    ```bash
+    $JAVA_HOME/bin/javac SVMHeapDump.java
+    ```
+
+2. Build a native executable using the `--enable-monitoring=heapdump` command-line option:
+    ```shell
+    $JAVA_HOME/bin/native-image SVMHeapDump --enable-monitoring=heapdump
+    ```
+    This causes the resulting native executable to produce a heap dump when it receives a `SIGUSR1` signal.
+
+    The `native-image` builder creates a native executable from the `SVMHeapDump.class`.
+    When the command completes, the native executable `svmheapdump` is created in the current directory.)
+
+### Run the Applicatoin
+
+1. Run the application send it a signal, and check the heap dump.:
+    ```shell
+    ./svmheapdump
+    ```
+    ```
+    17 May 2022, 16:38:13: Hello GraalVM native image developer! 
+    The PID of this process is: 57509
+    Send it a signal: 'kill -SIGUSR1 57509' 
+    to dump the heap into the working directory.
+    Starting thread!
+    17 May 2022, 16:38:13: Thread started, it will run for 60 seconds
+    ```
+    Make a note of the PID and open a second terminal. Use the PID to send a signal to the application. For example, if the PID is `57509`:
+    ```shell
+    kill -SIGUSR1 57509
+    ```
+    The heap dump will be created in the working directory while the application continues to run. The heap dump can be opened with the [VisualVM](https://www.graalvm.org/latest/tools/visualvm/) tool, as illustrated below.
+
+    ![Native Image Heap Dump View in VisualVM](img/heap-dump.png)
+
+## Create a Heap Dump from within a Native Executable
+
+The following example shows how to create a heap dump from a running native executable using [`VMRuntime.dumpHeap()`](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/VMInspection.java) if some condition is met.
+The condition to create a heap dump is provided as an option on the command line.
+1. Compile _SVMHeapDumpAPI.java_:
+    ```shell
+    $JAVA_HOME/bin/javac SVMHeapDumpAPI.java
+    ```
+    As in the earlier example, the application creates a `Collection` of `Person`s referenced by the static variable `CROWD`. It then checks the command line to see if heap dump has to be created, and then in method `createHeapDump()` creates the heap dump.
+
+2. Build a native executable:
+    ```bash
+    $JAVA_HOME/bin/native-image SVMHeapDumpAPI
+    ```
+    When the command completes, the `svmheapdumpapi` native executable is created in the current directory.
+
+3. Run the application and check the heap dump
```

Comment:
This and the sentence below tell to run the executable. Have just this:

3. Now run your native executable and create a heap dump from it with output similar to the following: